src/components/cluster/__tests__/cluster_managament.js

```diff
@@ -0,0 +1,106 @@
+import {
+  clusterCreate as mockClusterCreate,
+  clusterDelete as mockClusterDelete,
+} from 'actions/clusterActions';
+import { renderRouteWithStore } from 'test_utils/renderRouteWithStore';
+import initialState from 'test_utils/initialState';
+
+// Mock actions to return nothing, we don't want them perform API calls and we don't
+// want them to return any values either cause we are using a mocked store.
+jest.mock('actions/userActions');
+jest.mock('actions/organizationActions');
+jest.mock('actions/clusterActions', () => {
+  return {
+    clustersLoad: jest.fn(() => () => Promise.resolve()),
+    clusterLoadDetails: jest.fn(() => () => Promise.resolve()),
+    clusterLoadKeyPairs: jest.fn(() => () => Promise.resolve()),
+    clusterCreate: jest.fn(() => () => Promise.resolve()),
+    clusterDelete: jest.fn(() => () => Promise.resolve()),
+  };
+});
+jest.mock('actions/releaseActions');
+jest.mock('actions/catalogActions');
+
+// TODO find a way tot test router links
+it.skip('drives us to the cluster creation form when launch new cluster button is clicked', async () => {
+  const div = document.createElement('div');
+  const { getByText } = renderRouteWithStore('/', div);
+
+  await wait(() => {
+    const button = getByText(/launch new cluster/i);
+    fireEvent.click(button);
+    expect(getByText(/create cluster/i)).toBeInTheDocument();
+  });
+});
+
+it(`renders the form when in new cluster route with default values and calls
+    the action creator with the correct arguments`, async () => {
+  const div = document.createElement('div');
+  const { container } = renderRouteWithStore(
+    '/organizations/acme/clusters/new/',
+    div
+  );
+
+  await wait(() => container.querySelector('input'));
+
+  // TODO we need labels in the component instead h3s or alt attr on inputs,
+  // something more semantic than this. If
+  const inputs = container.querySelectorAll('input');
+  const name = inputs[0].value;
+  const availability_zones = +inputs[1].value;
+  const scaling = { min: +inputs[3].value, max: +inputs[4].value };
+  const owner = initialState.app.selectedOrganization;
+
+  const releases = initialState.entities.releases.items;
+  const releases_active = Object.keys(releases).filter(release => {
+    return releases[release].active === true;
+  });
+  const release_version = releases_active[releases_active.length - 1];
+  const workers = [
+    {
+      aws: {
+        instance_type: `${inputs[2].value}`,
+      },
+    },
+  ];
+
+  expect(inputs[0]).toHaveValue('My cluster');
+  expect(inputs[1]).toHaveValue(1);
+  expect(inputs[2]).toHaveValue('m3.large');
+  expect(inputs[3]).toHaveValue(3);
+  expect(inputs[4]).toHaveValue(3);
+
+  const submitButton = container.querySelector('button[type="submit"]');
+  fireEvent.click(submitButton);
+
+  const payload = {
+    availability_zones,
+    scaling,
+    name,
+    owner,
+    release_version,
+    workers,
+  };
+
+  expect(mockClusterCreate).toHaveBeenCalledTimes(1);
```

Comment:
I'd rather see here an expectation that we navigated to the page we expect, and that we see the title of the cluster as provided by the user and other expected details.

So that would be a matter of adding some details to the mocked `clusterCreate` call up there, taking the params from the payload, and dispatching a valid `CLUSTER_CREATE_SUCCESS` action with the parameters that the reducer expects.

Ideally though, if we could mock at the API response level, and not at the action level, that would be even better for this type of test.

Otherwise, what are we testing exactly here? That this button calls ends up calling an action with a certain payload? I think that is heavily in the realm of implementation detail, and not the right thing to be testing given the way that this test is set up.

Reply:
Mmmmh... you are right!  😄 

It's just that with the previous router setup I wasn't able to click and push a route. Now we maybe can do that... I will give it a try tomorrow.

Thanks!

Reply:
Not merging for now

Reply:
I have added a `statusState` object to inject when running the scaling test. I'm adding it to the `initialState` object.

And also I modified `node_count_selector`, now `<label>`s wrap `<input>`s

Reply:
ping @oponder You approved this, but as I did some modifications, asking you again for approval